engine/userguide/storagedriver/device-mapper-driver.md

```diff
@@ -504,196 +466,303 @@ The `Data Space` values show that the pool is 100GB total. This example extends
 
-    a.  Get the pool name first.
+        ```bash
+        $ sudo dmsetup table docker-8:1-123141-pool
+
+        0 209715200 thin-pool 7:1 7:0 128 32768 1 skip_block_zeroing
+        ```
+
+    c.  Calculate the total sectors of the thin pool using the second field
+        of the output. The number is expressed in 512-k sectors. A 100G file has
+        209715200 512-k sectors. If you double this number to 200G, you get
+        419430400 512-k sectors.
+
+    d.  Reload the thin pool with the new sector number, using the following
+        three `dmsetup`  commands.
+
+        ```bash
+        $ sudo dmsetup suspend docker-8:1-123141-pool
+
+        $ sudo dmsetup reload docker-8:1-123141-pool --table '0 419430400 thin-pool 7:1 7:0 128 32768 1 skip_block_zeroing'
+
+        $ sudo dmsetup resume docker-8:1-123141-pool
+        ```
+
+#### Resize a direct-lvm thin pool
+
+To extend a `direct-lvm` thin pool, you need to first attach a new block device
+to the Docker host, and make note of the name assigned to it by the kernel. In
+this example, the new block device is `/dev/xvdg`.
+
+Follow this procedure to extend a `direct-lvm` thin pool, substituting your
+block device and other parameters to suit your situation.
+
+1.  Gather information about your volume group.
+
+    Use the `pvdisplay` command to find the physical block devices currently in
+    use by your thin pool, and the volume group's name.
 
     ```bash
-    $ sudo dmsetup status | grep pool
+    $ sudo pvdisplay |grep 'VG Name'
 
-    docker-8:1-123141-pool: 0 209715200 thin-pool 91
-    422/524288 18338/1638400 - rw discard_passdown queue_if_no_space -
+    PV Name               /dev/xvdf
+    VG Name               docker
     ```
 
-    The name is the string before the colon.
+    In the following steps, substitute your block device or volume group name as
+    appropriate.
 
-    b.  Dump the device mapper table first.
+2.  Extend the volume group, using the `vgextend` command with the `VG Name`
+    from the previous step, and the name of your **new** block device.
 
     ```bash
-    $ sudo dmsetup table docker-8:1-123141-pool
+    $ sudo vgextend docker /dev/xvdg
 
-    0 209715200 thin-pool 7:1 7:0 128 32768 1 skip_block_zeroing
+    Physical volume "/dev/xvdg" successfully created.
+    Volume group "docker" successfully extended
     ```
 
-    c.  Calculate the real total sectors of the thin pool now.
+3.  Extend the `thinpool/data` logical volume. This command uses 100% of the
+    volume right away, without auto-extend. If you needed to extend the
+    `thinpool/metadata` logical volume instead, substitute it.
 
-        Change the second number of the table info (i.e. the disk end sector) to
-        reflect the new number of 512 byte sectors in the disk. For example, as the
-        new loop size is 200GB, change the second number to 419430400.
+    ```bash
+    $ sudo lvextend  -l+100%FREE -n thinpool/data
```

Comment:
This command doesn't work with the `docker/thinpool` naming convention:

```
$ sudo lvresize -l+100%FREE -n thinpool/data
  Volume group "thinpool" not found
  Cannot process volume group thinpool
```

Correcting for the naming convention, this would be:

```
$ sudo lvextend -l+100%FREE -n docker/thinpool_tdata
  Can't resize internal logical volume docker/thinpool_tdata.
```

But that also fails, so the intent is probably:

```
$ sudo lvextend -l+100%FREE -n docker/thinpool
  Size of logical volume docker/thinpool_tdata changed from 100.00 GiB (25599 extents) to 195.00 GiB (49921 extents).
  Logical volume docker/thinpool_tdata successfully resized.
```